cpp/src/arrow/vendored/fast_float/parse_number.h

```diff
@@ -30,6 +30,9 @@ from_chars_result parse_infnan(const char *first, const char *last, T &value)  n
       minusSign = true;
       ++first;
   }
+  if (*first == '+') {
```

Comment:
The comment at line 29 "C++17 20.19.3.(7.1) explicitly forbids '+' here" seems to be outdated

Reply:
not really, in fast_float version of 3.10.1( current is 3.8.1) https://github.com/fastfloat/fast_float/blob/main/include/fast_float/parse_number.h

was added 

```
#ifdef FASTFLOAT_ALLOWS_LEADING_PLUS // disabled by default
  if (*first == UC('+')) {
      ++first;
  }
```

I created pr in apache arrow for updating fast_float version but i don't know when it will merged or will it be merged at all. So for now we could use this.

Reply:
and i don't want to remove any comments because it's a custom lib, and will be conflict on merging